source/elements/oneDAL/source/glossary.rst

```diff
@@ -0,0 +1,159 @@
+.. _glossary:
+
+=========
+Glossary
+=========
+
+Machine learning terms
+======================
+
+.. glossary::
+    :sorted:
+
+    Categorical feature
+        A :term:`feature` with a discrete set of possible values without ordering.
+
+        Synonyms: nominal feature.
+
+    Classification
+        A :term:`supervised machine-learning problem<Supervised learning>` of assigning
+        discrete :term:`labels<Label>` (:term:`categorical<Categorical feature>` or :term:`ordinal<Ordinal feature>`)
+        to :term:`feature vectors<Feature vector>`.
+
+    Continuous feature
+        A :term:`feature` with values in a domain of real numbers.
+
+        Synonyms: quantitative feature.
+
+    Dataset
+        A collection of :term:`observations<Observation>`.
+
+    Feature
```

Comment:
makes sense to link to the features of different types

and a general q: do we maybe want to list related terms? If something that might be relevant for the reader but not specifically appears in the definition

Reply:
I would not want to do this, because the glossary transforms into an ML guide. In my opinion, the terms should be short, like in vocabulary. The reader must be referenced to the **direct** term he wants to clarify in the spec. And this term can refer to other terms. 

@rlnx @andreynv what is your opinion?

Reply:
the quick example from the top of my mind. During the design of dbscan the terminology that came from the paper was 'points' or so which means observation of feature vector for the lib. To make sure that the dbscan is described in properly, we introduced that synonym.  to simplify, we might have the same term and rely on it across the specs even if it would deviate from the terminology in the paper (for 'dbscan' we might refer to the paper and mention that it uses 'point' instead of 'observation')